### PR TITLE
Decompile mon markings, rotating_gate & field_camera

### PR DESCRIPTION
**Pull request by KDSKardabox**

### CODE REVIEW

File: src/mon_markings.c [hunk 1]
```diff
@@ -0,0 +1,616 @@
...
+#include "main.h"
+#include "menu_indicators.h"
+#include "mon_markings.h"
+#include "constants/songs.h"
+#include "sound.h"
+#include "sprite.h"
+#include "text_window.h"
+
+
+#ifdef GERMAN
```
Comment (huderlem) [inline src/mon_markings.c]: Pokeemerald doesn't build German--this can be removed.

Comment (KDSKardabox) [reply src/mon_markings.c]: done

File: src/rotating_gate.c [hunk 2]
```diff
@@ -0,0 +1,1195 @@
...
+
+    return PUZZLE_NONE;
+}
+
+static void RotatingGate_ResetAllGateOrientations(void)
+{
+    s32 i;
+    u8 *ptr;
+
+    ptr = (u8 *)GetVarPointer(0x4000);
```
Comment (Diegoisawesome) [inline src/rotating_gate.c]: Use constant for 0x4000

File: src/rotating_gate.c [hunk 3]
```diff
@@ -0,0 +1,1195 @@
...
+
+static void RotatingGate_RotateInDirection(u8 gateId, u32 rotationDirection)
+{
+    u8 orientation = RotatingGate_GetGateOrientation(gateId);
+
+    if (rotationDirection == ROTATE_ANTICLOCKWISE)
+    {
+        if (orientation)
+        {
+            orientation--;
```
Comment (Diegoisawesome) [inline src/rotating_gate.c]: Both parts of if statement are single-line, remove curly braces

File: src/rotating_gate.c [hunk 4]
```diff
@@ -0,0 +1,1195 @@
...
+
+    sub_8092FF0(x + deltaX, y + deltaY, &sprite->pos1.x, &sprite->pos1.y);
+    RotatingGate_HideGatesOutsideViewport(sprite);
+    StartSpriteAffineAnim(sprite, RotatingGate_GetGateOrientation(gateId));
+
+    return spriteId;
+}
+
+/*
+static*/ void SpriteCallback_RotatingGate(struct Sprite *sprite)
```
Comment (Diegoisawesome) [inline src/rotating_gate.c]: Is it actually static? Either delete or uncomment

File: src/rotating_gate.c [hunk 5]
```diff
@@ -0,0 +1,1195 @@
...
+
+        PlaySE(SE_HI_TURUN);
+        StartSpriteAffineAnim(sprite, affineAnimation);
+    }
+
+    sprite->data[1] = ROTATE_NONE;
+}
+
+/*
+static*/ void RotatingGate_HideGatesOutsideViewport(struct Sprite *sprite)
```
Comment (Diegoisawesome) [inline src/rotating_gate.c]: Same here

**Pull request merged**